perun-openapi/openapi.yml

```diff
@@ -8140,6 +8206,241 @@ paths:
+                candidate: { $ref: '#/components/schemas/Candidate' }
+                groups: { type: array, items: { $ref: '#/components/schemas/Group' } }
+
+
+  /json/membersManager/createMember/u:
+    post:
+      tags:
+        - MembersManager
+      operationId: CreateMemberForUser
+      summary: Creates a new member from user. This method also add user to all groups in list.
+        Empty list of groups is ok, the behavior is then same like in the method without list of groups.
+      responses:
+        '200':
+          $ref: '#/components/responses/MemberResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              title: InputCreateMemberForUser
+              description: "input to create member for user"
+              type: object
+              required:
+                - vo
+                - user
+              properties:
+                vo: { type: integer }
+                user: { type: integer }
+                groups: { type: array, items: { $ref: '#/components/schemas/Group' } }
+
+  /json/membersManager/createMember/c:
+    post:
+      tags:
+        - MembersManager
+      operationId: createMemberForCandidate
+      summary: Creates a new member from candidate returned by the method VosManager.findCandidates
+        which fills Candidate.userExtSource. This method also add user to all groups in list. Empty list of groups is ok,
+        the behavior is then same like in the method without list of groups. This method runs asynchronously
+      responses:
+        '200':
+          $ref: '#/components/responses/MemberResponse'
+        default:
+          $ref: '#/components/responses/ExceptionResponse'
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              title: InputCreateMemberForCandidate
+              description: "input to create member for candidate"
+              type: object
+              required:
+                - vo
+                - candidate
+              properties:
+                vo: { type: integer }
+                candidate: { $ref: '#/components/schemas/Candidate' }
+                groups: { type: array, items: { $ref: '#/components/schemas/Group' } }
+
+  /json/membersManager/getCompleteRichMembers/v:
+    get:
+      tags:
+        - MembersManager
+      operationId: getCompleteRichMembersForVo
+      summary: Get all RichMembers with attributes specific for list of attrsNames from the vo and have only status
+        which is contain in list of statuses. If attrsNames is empty or null return all attributes for specific richMembers.
+        If listOfStatuses is empty or null, return all possible statuses.
+      parameters:
+        - $ref: '#/components/parameters/voId'
+        - $ref: '#/components/parameters/listOfAttributesNames'
```

Comment:
This referenced attribute is set as required but I think for this scenario it should be optional.

Reply:
You can use `attrsNames` which is already declared as a parameter but you have to fix its name. I have noticed it is wrong `attrsNames` -> `attrsNames[]`